### PR TITLE
NLRFIM-125 - Missing KeyInfo in XML signature

### PR DESCRIPTION
**Pull request by TauBeblein**
SSO issue with logout response: https://www.digitaliser.dk/forum/6202895

Ready for review

### CODE REVIEW

File: idp/src/main/java/dk/itst/oiosaml/idp/controller/saml/SingleSignOnController.java [hunk 1]
```diff
@@ -14,7 +14,7 @@
 import org.opensaml.saml.common.messaging.context.SAMLEndpointContext;
 import org.opensaml.saml.common.messaging.context.SAMLPeerEntityContext;
 import org.opensaml.saml.common.xml.SAMLConstants;
-import org.opensaml.saml.saml2.binding.encoding.impl.HTTPPostEncoder;
+import org.opensaml.saml.saml2.binding.encoding.impl.HTTPPostSimpleSignEncoder;
```
Comment (mdbech) [inline idp/src/main/java/dk/itst/oiosaml/idp/controller/saml/SingleSignOnController.java:17]: Didn't we reach the conclusion that HTTPPostSimpleSignEncoder was wrong? We don't want to double sign the content.

File: oiosaml/src/main/java/dk/gov/oio/saml/servlet/SAMLHandler.java [hunk 2]
```diff
@@ -86,7 +86,7 @@ void sendPost(HttpServletResponse httpServletResponse, MessageContext<SAMLObject
             log.debug("Encoding and sending message (HTTPPost)");
         }
 
-        HTTPPostEncoder encoder = new HTTPPostEncoder();
+        HTTPPostSimpleSignEncoder encoder = new HTTPPostSimpleSignEncoder();
```
Comment (mdbech) [inline oiosaml/src/main/java/dk/gov/oio/saml/servlet/SAMLHandler.java:89]: Same here

File: oiosaml/src/test/java/dk/gov/oio/saml/servlet/LogoutRequestHandlerTest.java [hunk 3]
```diff
@@ -116,17 +113,39 @@ public void testIdPLogoutRequestWhenLoggedIn() throws Exception {
 		Mockito.when(request.getParameter("SAMLRequest")).thenReturn(base64EncodedMessage);
 
 		// Mock DummyOutputStream
-		DummyOutputStream dummyOutputStream = Mockito.mock(DummyOutputStream.class);
+		DummyOutputStream dummyOutputStream = Mockito.spy(new DummyOutputStream());
```
Comment (mdbech) [inline oiosaml/src/test/java/dk/gov/oio/saml/servlet/LogoutRequestHandlerTest.java:116]: @sihpe or @jpramming do you have an opinion on mock vs spy?

**Review by mdbech — CHANGES_REQUESTED** [review]

File: oiosaml/src/test/java/dk/gov/oio/saml/servlet/LogoutRequestHandlerTest.java (hunk 3 above)
Comment (TauBeblein) [reply oiosaml/src/test/java/dk/gov/oio/saml/servlet/LogoutRequestHandlerTest.java:116]: Spy is used to enabling validation of content on the output stream

File: idp/src/main/java/dk/itst/oiosaml/idp/controller/saml/SingleSignOnController.java (hunk 1 above)
Comment (TauBeblein) [reply idp/src/main/java/dk/itst/oiosaml/idp/controller/saml/SingleSignOnController.java]: done

File: oiosaml/src/main/java/dk/gov/oio/saml/servlet/SAMLHandler.java (hunk 2 above)
Comment (TauBeblein) [reply oiosaml/src/main/java/dk/gov/oio/saml/servlet/SAMLHandler.java]: done

File: oiosaml/src/test/java/dk/gov/oio/saml/servlet/LogoutRequestHandlerTest.java (hunk 3 above)
Comment (TauBeblein) [reply oiosaml/src/test/java/dk/gov/oio/saml/servlet/LogoutRequestHandlerTest.java]: There is no longer a reason to validate generated HTML so cleaned up the code

**Review by mdbech — APPROVED** [review]

**Pull request merged**